kernel/src/table_changes/scan.rs

```diff
@@ -148,15 +151,38 @@ impl TableChangesScanBuilder {
                 }
             })
             .try_collect()?;
+        let table_schema = self.table_changes.end_snapshot.schema().clone().into();
         Ok(TableChangesScan {
             table_changes: self.table_changes,
             logical_schema,
             predicate: self.predicate,
             all_fields,
             have_partition_cols,
+            physical_schema: StructType::new(read_fields),
+            table_schema,
         })
     }
 }
+
+impl TableChangesScan {
+    pub fn scan_data(
+        &self,
+        engine: &dyn Engine,
+    ) -> DeltaResult<impl Iterator<Item = DeltaResult<TableChangesScanData>>> {
+        let commits = self
```

Comment:
Just wanted to make sure that copying `ParsedLogPath` is okay.  Passing a `&[&ParsedLogPath]` makes it so the LogReplayScanner holds it, and the iterator now has a lifetime tied to `table_changes.log_segment`.

Reply:
It's fairly small, so doing it once per query is likely fine